pkg/native/serializers/serializer_cdx_test.go

```diff
@@ -190,3 +192,240 @@ func TestPurposeToComponentType(t *testing.T) {
+					NodeList: &sbom.NodeList{
+						RootElements: []string{"root1", "root2"},
+					},
+				},
+				in1: nil,
+				in2: nil,
+			},
+			want:    nil,
+			wantErr: true,
+		},
+
+		{
+			name: "Valid BOM with Single Root Node",
+			fields: fields{
+				version:  "1.4",
+				encoding: "json",
+			},
+			args: args{
+				bom: &sbom.Document{
+					Metadata: &sbom.Metadata{
+						Id:      "1234",
+						Version: "1",
+					},
+					NodeList: &sbom.NodeList{
+						RootElements: []string{"root1"},
+						Nodes: []*sbom.Node{
+							{Id: "root1", Name: "Root Node"},
+						},
+					},
+				},
+				in1: nil,
+				in2: nil,
+			},
+			want: &cdx.BOM{
+				XMLNS:        "http://cyclonedx.org/schema/bom/1.5",
+				JSONSchema:   "http://cyclonedx.org/schema/bom-1.5.schema.json",
+				BOMFormat:    "CycloneDX",
+				SpecVersion:  cdx.SpecVersion(6),
+				SerialNumber: "1234",
+				Version:      1,
+				Metadata: &cdx.Metadata{
+					Component: &cdx.Component{
+						BOMRef:             "root1",
+						Name:               "Root Node",
+						Hashes:             &[]cdx.Hash{},
+						ExternalReferences: &[]cdx.ExternalReference{},
+						Type:               "",
+					},
+					Lifecycles: &[]cdx.Lifecycle{},
+				},
+				Components:   &[]cdx.Component{},
+				Dependencies: &nilSlice,
+			},
+			wantErr: false,
+		},
+		{
+			name: "Valid filled out BOM",
+			fields: fields{
+				version:  "1.4",
+				encoding: "json",
+			},
+			args: args{
+				bom: &sbom.Document{
+					Metadata: &sbom.Metadata{
+						Id:      "1234",
+						Version: "1",
+						DocumentTypes: []*sbom.DocumentType{
+							{
+								Type: sbom.DocumentType_ANALYZED.Enum(),
+								Name: func() *string {
+									name := "Analyzed Document"
+									return &name
+								}(),
+							},
+						},
+						Authors: []*sbom.Person{
+							{
+								Name:  "TestName",
+								Email: "TestEmail",
+								Phone: "TestPhone",
+							},
+						},
+						Tools: []*sbom.Tool{
+							{
+								Name:    "ToolName",
+								Version: "2",
+							},
+						},
+						Name: "DocName",
+					},
+					NodeList: &sbom.NodeList{
+						RootElements: []string{"root1"},
+						Nodes: []*sbom.Node{
+							{Id: "root1", Name: "Root Node 1"},
+						},
+					},
+				},
+				in1: nil,
+				in2: nil,
+			},
+			want: &cdx.BOM{
+				XMLNS:        "http://cyclonedx.org/schema/bom/1.5",
+				JSONSchema:   "http://cyclonedx.org/schema/bom-1.5.schema.json",
+				BOMFormat:    "CycloneDX",
+				SpecVersion:  cdx.SpecVersion(6),
+				SerialNumber: "1234",
+				Version:      1,
+				Metadata: &cdx.Metadata{
+					Component: &cdx.Component{
+						BOMRef:             "root1",
+						Name:               "DocName",
+						Hashes:             &[]cdx.Hash{},
+						ExternalReferences: &[]cdx.ExternalReference{},
+						Type:               "",
+					},
+					Lifecycles: &[]cdx.Lifecycle{{Phase: cdx.LifecyclePhasePostBuild}},
+					Tools:      &cdx.ToolsChoice{Tools: &[]cdx.Tool{{Version: "2", Name: "ToolName"}}},
```

Comment:
@neilnaveen feel free to skip the linter here, there is an unresolved issue on how to handle the component subtree introduced in CDX 1.5

Reply:
Thanks for the review. I have fixed the lint error.